website/content/tutorials/2-constraints/_index.md

```diff
@@ -0,0 +1,1069 @@
+      speed: 4.7 gigahertz
+    ata-socket:
+      vendor:
+        id: vendor2
+        name: ISA Corp
+        address: 2000 K Street NW Washington, DC 20002
+        website: https://example.com/isacorp/
+      product-name: SuperSATA Model 2 Storage Socket
+      type: sata
+    memory-modules:
+    - vendor:
+        id: vendor3
+        name: Massive Memory Corp
+        address: 3000 K Street NW Washington, DC 20003
+        website: https://example.com/massive-memory-corp/
+      product-name: Model 3 DDR4-3200 8GB (Module 1)
+      byte-size: 8589934592
+    - byte-size: 8589934592
+      product-name: Model 3 DDR4-3200 8GB (Module 2)
+      vendor:
+        address: 3000 K Street NW Washington, DC 20003
+        id: vendor3
+        name: Massive Memory Corp
+        website: https://example.com/massive-memory-corp/
+{{< /highlight >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+### Recommended values with `allow-other="yes"`
+
+The updated model with approved values for CPU architecture was a success with stakeholders. It allowed those using Metaschema-enabled tools to exchange more consistent, accurate data. Now the computer model stakeholders have a new requirement. For the ATA socket type for the motherboard of a computer, stakeholders want to have a set of approved values. Unlike the CPU architecture type, the stakeholders want to maintain flexibility for emerging socket technologies without implementations or names at present. At a minimum, they want the preferred values in the table below to be recommended, but allow alternatives as they emerge.
+
+| ATA Socket Type   | Description                                                                        |
+|-------------------|------------------------------------------------------------------------------------|
+| pata              | Parallel ATA buses also known as AT-Attachment and IDE                             |
+| sata              | Serial ATA buses supporting Advanced Host Controller Interface or legacy IDE modes |
+| esata             | External Serial ATA buses for pluggable external devices using SATA                |
+| esatap            | External Serial ATA buses supporting SATA traffic and device power                 |
+
+To implement such a requirement, we must use an [`<allowed-values>` constraint](/specification/syntax/constraints#enumerated-values) to make a list of optional, recommended values. For a field like architecture in the model, we can define it inside the field like below where the change is highlighted. The `allow-other="yes"` declaration means a valid, conforming document instance can use a recommended value from the list, or any other as new types emerge. The constraint definition is inside the architecture field so it is the current context for evaluation focus of the [target](/specification/syntax/constraints#target). The declaration `target="."` makes it explicit.
+
+```xml {linenos=table,hl_lines=["143-154"]}
```

Comment:
non-blocking - Should these examples be cumulative? Should the CPU architecture be included in the following example? Since we are highlighting the new change, I don't there isn't much of a downside.

Reply:
Will do, I think that is worthwhile.

Reply:
Updated in https://github.com/usnistgov/metaschema/pull/508/commits/889758b6fe05bdbb6c5dd410dc00aabe7d95b787.